backend/tracim_backend/lib/mail_fetcher/email_processing/sanitizer.py

```diff
@@ -28,41 +37,53 @@ class HtmlSanitizer(object):
       - Remove non-whitelisted attributes
     """
 
-    @classmethod
-    def sanitize(cls, html_body: str) -> typing.Optional[str]:
-        soup = BeautifulSoup(html_body, 'html.parser')
-        for tag in soup.findAll():
-            if cls._tag_to_extract(tag):
+    def __init__(self, html_body: str, config: HtmlSanitizerConfig = None):
+        if config is None:
+            config = HtmlSanitizerConfig()
+        self.config = config
+        self.soup = BeautifulSoup(html_body, config.parser)
+
+    def sanitize(self) -> typing.Optional[str]:
+        for tag in self.soup.findAll():
+            if self._tag_to_extract(tag):
                 tag.extract()
-            elif tag.name.lower() in HtmlSanitizerConfig.Tag_whitelist:
+            elif tag.name.lower() in self.config.tag_whitelist:
                 attrs = dict(tag.attrs)
                 for attr in attrs:
-                    if attr not in HtmlSanitizerConfig.Attrs_whitelist:
+                    if attr not in self.config.attrs_whitelist:
                         del tag.attrs[attr]
             else:
                 tag.unwrap()
 
-        if cls._is_content_empty(soup):
+        if self._is_content_empty(self.soup):
             return None
         else:
-            return str(soup)
+            return str(self.soup)
 
-    @classmethod
-    def _is_content_empty(cls, soup):
+    def _is_content_empty(self, soup: Tag):
         img = soup.find('img')
         txt = soup.get_text().replace('\n', '').strip()
         return (not img and not txt)
 
-    @classmethod
-    def _tag_to_extract(cls, tag: Tag) -> bool:
-        if tag.name.lower() in HtmlSanitizerConfig.Tag_blacklist:
+    def _tag_to_extract(self, tag: Tag) -> bool:
+        if tag.name.lower() in self.config.tag_blacklist:
             return True
         if 'class' in tag.attrs:
-            for elem in HtmlSanitizerConfig.Class_blacklist:
+            for elem in self.config.class_blacklist:
                 if elem in tag.attrs['class']:
                     return True
         if 'id' in tag.attrs:
-            for elem in HtmlSanitizerConfig.Id_blacklist:
+            for elem in self.config.id_blacklist:
                 if elem in tag.attrs['id']:
                     return True
         return False
+
+    def html_is_empty(self, soup: Tag = None) -> bool:
+        if not soup:
+            soup = self.soup
+        if not self._is_content_empty(self.soup):
+            return False
+        return all((self.html_is_empty(child) for child in soup.children))
+
+    def is_html(self):
```

Comment:
docstring missing and return type missing

Reply:
done